### PR TITLE
feat: add binary format to releaser

### PR DESCRIPTION
**Pull request by RafilxTenfen**
## Description

- Release directly binaries instead of tarballs to make it compatible with cosmovisor
- The current `goreleaser` is going to generate an release like this: https://github.com/RafilxTenfen/umee/releases/tag/v3.1.1-rc0

closes: #1347 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by RafilxTenfen** [comment]
I am thinking how can I test that :thinking:

**Comment by RafilxTenfen** [comment]
doc: https://goreleaser.com/customization/archive/?h=archives

File: .goreleaser.yml [hunk 1]
```diff
@@ -28,11 +28,7 @@ builds:
       - arm64
 
 archives:
-  - format: tar.gz
```
Comment (robert-zaremba) [inline .goreleaser.yml:31]: let's use `zip` instead of `tar.gz`. `zip` works with cosmovisor.

Comment (RafilxTenfen) [reply .goreleaser.yml:31]: nice, I will try to run it locally with zip tarballs

Comment (RafilxTenfen) [reply .goreleaser.yml:31]: did not work :/ 
https://github.com/umee-network/umee-infra/pull/263#issuecomment-1250414590

Comment (RafilxTenfen) [reply .goreleaser.yml:31]: @robert-zaremba can you check this again?

**Comment by RafilxTenfen** [comment]
Tested again against [real release](https://github.com/RafilxTenfen/umee/releases/tag/v3.1.0-rc9) files (tarball and plain binaries)
- https://github.com/umee-network/umee-infra/pull/263 only worked with plain binaries  

For this PR and to maintain the tarballs I will set the `goreleaser` to generate 2 archives, the plain binaries which we should point to work with [cosmovisor](https://github.com/cosmos/cosmos-sdk/blob/main/cosmovisor/README.md), and the tarballs to keep what we currently have

**Review by adamewozniak — APPROVED** [review]
Approving, might want another look from @robert-zaremba first. If we can prove that it'll work with our tarballs then we can block

**Review by toteki — APPROVED** [review]
Looks good - and since this is just an addition to the release (no code / potential bugs), we should get it in before 3.1.0

**Comment by RafilxTenfen** [comment]
Tested again and now was able to make it work with tarballs (tar.gz or zip), the thing is we need to remove `wrap_in_directory` and in the cosmovisor set `DAEMON_NAME=umeed` as env 

but, if we use the binary release link directly, there is no need to set  `DAEMON_NAME`

**Pull request merged**